tests/ArrayCollectionTestCase.php

```diff
@@ -457,4 +458,48 @@ public function testMultiColumnSortAppliesAllSorts(): void
                 ->toArray(),
         );
     }
+
+    public function testMultiColumnSortAppliesAllSortsAndTreatsDateTimeInstancesAsValues(): void
+    {
+        $collection = $this->buildCollection([
+            ['foo' => new DateTimeImmutable('2025-01-01 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-02 12:00:00')],
+            ['foo' => new DateTimeImmutable('2025-01-01 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-04 12:00:00')],
+            ['foo' => new DateTimeImmutable('2025-01-01 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-03 12:00:00')],
+            ['foo' => new DateTimeImmutable('2025-01-02 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-05 12:00:00')],
+        ]);
+
+        $expected = [
+            3 => ['foo' => new DateTimeImmutable('2025-01-02 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-05 12:00:00')],
+            0 => ['foo' => new DateTimeImmutable('2025-01-01 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-02 12:00:00')],
+            2 => ['foo' => new DateTimeImmutable('2025-01-01 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-03 12:00:00')],
+            1 => ['foo' => new DateTimeImmutable('2025-01-01 12:00:00'), 'bar' => new DateTimeImmutable('2025-01-04 12:00:00')],
+        ];
+
+        if (! $this->isSelectable($collection)) {
+            $this->markTestSkipped('Collection does not support Selectable interface');
+        }
+
+        $actual = $collection
+            ->matching((new Criteria(null, ['foo' => Order::Descending, 'bar' => Order::Ascending]))->treatDateTimeAsScalar())
```

Comment:
I would expect this call to result in `treatDateTimeAsScalar` being covered. :thinking:

Reply:
Oh, I guess this explains it: https://github.com/doctrine/collections/blob/793b4dfd3252dab123f5adb35e55474602ec4d23/tests/ArrayCollectionTest.php#L16